### ISSUE TITLE
Compile fails under arm64

### ISSUE THREAD

**Issue by yagop** [near/nearcore#6281]
**Describe the bug**
Can't compile under arm64
```
   Compiling wasmtime-jit v0.33.0
error[E0433]: failed to resolve: use of undeclared crate or module `rustix`
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/wasmtime-jit-0.33.0/src/code_memory.rs:60:13
   |
60 |             rustix::process::membarrier(
   |             ^^^^^^ use of undeclared crate or module `rustix`
error[E0433]: failed to resolve: use of undeclared crate or module `rustix`
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/wasmtime-jit-0.33.0/src/code_memory.rs:61:17
   |
61 |                 rustix::process::MembarrierCommand::RegisterPrivateExpeditedSyncCore,
   |                 ^^^^^^ use of undeclared crate or module `rustix`
error[E0433]: failed to resolve: use of undeclared crate or module `rustix`
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/wasmtime-jit-0.33.0/src/code_memory.rs:174:17
    |
174 |                 rustix::process::membarrier(
    |                 ^^^^^^ use of undeclared crate or module `rustix`
error[E0433]: failed to resolve: use of undeclared crate or module `rustix`
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/wasmtime-jit-0.33.0/src/code_memory.rs:175:21
    |
175 |                     rustix::process::MembarrierCommand::PrivateExpeditedSyncCore,
    |                     ^^^^^^ use of undeclared crate or module `rustix`
```

**To Reproduce**
```
FROM ubuntu:18.04 as build

RUN apt-get update -qq && apt-get install -y \
    git \
    cmake \
    g++ \
    pkg-config \
    libssl-dev \
    curl \
    llvm \
    clang \
    unzip \
    && rm -rf /var/lib/apt/lists/*

ENV NEAR_VERSION="1.23.1"
ENV NEAR_COMMIT="9110028c663b5f9c1a353e064a535f0ebedd82e0"

# https://github.com/near/nearcore/archive/refs/tags/1.23.1.tar.gz
# ADD https://github.com/near/nearcore/archive/refs/tags/${NEAR_VERSION}.tar.gz /tmp/nearcore-${NEAR_VERSION}.tar.gz
# RUN tar xf /tmp/nearcore-${NEAR_VERSION}.tar.gz \
#    && mv ./nearcore-${NEAR_VERSION} /tmp/near

ADD https://github.com/near/nearcore/archive/${NEAR_COMMIT}.zip /tmp/nearcore.zip
RUN unzip /tmp/nearcore.zip \
    && mv ./nearcore-${NEAR_COMMIT} /tmp/near

COPY ./install-rust.sh /tmp/install-rust.sh

ENV RUSTUP_HOME="/usr/local/rustup" \
    CARGO_HOME="/usr/local/cargo" \
    PATH="/usr/local/cargo/bin:$PATH"
RUN cp /tmp/near/rust-toolchain.toml /tmp/rust-toolchain.toml \
    && /tmp/install-rust.sh \
    && rm /tmp/install-rust.sh

WORKDIR /tmp/near

ENV PORTABLE=ON

RUN CARGO_TARGET_DIR=/tmp/target cargo build -p neard --release --bin neard \
    && CARGO_TARGET_DIR=/tmp/sandbox cargo build -p neard --features sandbox --release
```
**Expected behavior**
Getting near compiled

**Screenshots**

**Version (please complete the following information):**
`9110028c663b5f9c1a353e064a535f0ebedd82e0` from #6237

- rust version:
1.58.1

**Comment by mikroskeem** [comment]
Can confirm.

**Comment by bowenwang1996** [comment]
@Ekleog can you confirm that this is already fixed in master?

**Comment by Ekleog** [comment]
@yagop Your dockerfile lists 1.23.1 as a neard version, which dates back to before neard was ported to aarch64. As for 1.25.0-rc.3, I have not tried with it because neard doesn't work on m1 there anyway, it only builds successfully — 1.25.0 was cut before https://github.com/near/nearcore/pull/6388 happened.

This being said, I've tried `docker build --platform="linux/arm64" --build-arg make_target="neard" .` on the latest head, and it appears to fail with the same issue here.

I'm not sure what the reason for this error is, and we do not officially support aarch64-linux-* targets yet, only aarch64 darwin. (running inside docker means running aarch64 ubuntu 18.04) Overall this looks to me like the issue is upstream at wasmtime-jit, but I may be wrong as I have not investigated more in depth yet.

For now, the workaround would be to not use docker: both `cargo build -p neard --release --bin neard` and `cargo build -p neard --features sandbox --release` both complete successfully on an m1 mac.

@bowenwang1996 Do we want to add aarch64-linux to the list of our partially-supported platforms? (this would be necessary in order to make docker work; but I thought we wanted to only support local development on macs and thus docker-based systems would afaict not be very meaningful for this use case)

**Comment by bowenwang1996** [comment]
> @bowenwang1996 Do we want to add aarch64-linux to the list of our partially-supported platforms? (this would be necessary in order to make docker work; but I thought we wanted to only support local development on macs and thus docker-based systems would afaict not be very meaningful for this use case)

Yeah let's postpone this for now since we only support local development for aarch64 right now

**Comment by cif** [comment]
We're using postgres to persist indexer data and our code won't compile on an M1 locally because libpq doesn't yet have a universal binary. This lead us to try docker and encountered the same problem above. 

More searching turned up an [issue that was fixed in a later version of wasmtime](https://github.com/bytecodealliance/wasmtime/issues/3672).  Seem on the surface that updating wasmtime dependencies could solve this?  I will see what I can do to fork and replicate as time allows, but I wanted to bring this up here in case someone could verify this might work or tell me I'm wrong.

**Comment by bowenwang1996** [comment]
> We're using postgres to persist indexer data and our code won't compile on an M1 locally because libpq doesn't yet have a universal binary. This lead us to try docker and encountered the same problem above.
> 
> More searching turned up an [issue that was fixed in a later version of wasmtime](https://github.com/bytecodealliance/wasmtime/issues/3672). Seems on the surface that updating wasmtime dependencies could solve this? I will see what I can do to fork and replicate as time allows, but I wanted to bring this up here in case someone could verify this might work or tell me I'm wrong.

@Ekleog what do you think?

**Comment by Ekleog** [comment]
Sorry for the delay in my answer, this totally slipped off my radar! So updating wasmtime should not be a problem in practice. However, due to the way actix is being developed, right now we need to wait until https://github.com/near/nearcore/pull/6694 lands before we can run cargo update (there's a PR ready at https://github.com/near/nearcore/pull/6712 ).

Once this will have been completed, we can go back to updating wasmtime :)

**Comment by cif** [comment]
No problem at all, thanks for the update!  I dug up my old 2013 MPB Intel that will compile our indexer for now.  Would be great to use the M1, we will track the PR.

**Comment by Ekleog** [comment]
I have just pushed #7086 with the wasmtime upgrade, if I understand bytecodealliance/wasmtime#3672 correctly it should fix this issue :)
